locked-asset/simple-lock-energy/src/token_merging.rs

```diff
@@ -0,0 +1,144 @@
+
+        self.token_amount += other.token_amount;
+        self.attributes.unlock_epoch = unsafe { new_unlock_epoch.to_u64().unwrap_unchecked() };
+    }
+}
+
+#[elrond_wasm::module]
+pub trait TokenMergingModule:
+    simple_lock::basic_lock_unlock::BasicLockUnlock
+    + simple_lock::locked_token::LockedTokenModule
+    + elrond_wasm_modules::default_issue_callbacks::DefaultIssueCallbacksModule
+    + simple_lock::token_attributes::TokenAttributesModule
+    + crate::util::UtilModule
+    + elrond_wasm_modules::pause::PauseModule
+    + crate::energy::EnergyModule
+    + crate::events::EventsModule
+    + crate::lock_options::LockOptionsModule
+{
+    #[endpoint(mergeTokens)]
+    fn merge_tokens(&self) -> EsdtTokenPayment {
+        self.require_not_paused();
+
+        let current_epoch = self.blockchain().get_block_epoch();
+        let caller = self.blockchain().get_caller();
+        let locked_token_mapper = self.locked_token();
+
+        let mut payments = self.get_non_empty_payments();
+        locked_token_mapper.require_all_same_token(&payments);
+
+        let first_payment = payments.get(0);
+        payments.remove(0);
+
+        let output_amount_attributes =
+            self.update_energy(&caller, |energy: &mut Energy<Self::Api>| {
+                let first_token_attributes: LockedTokenAttributes<Self::Api> =
+                    locked_token_mapper.get_token_attributes(first_payment.token_nonce);
+                energy.update_after_unlock_any(
+                    &first_payment.amount,
+                    first_token_attributes.unlock_epoch,
+                    current_epoch,
+                );
+
+                locked_token_mapper.nft_burn(first_payment.token_nonce, &first_payment.amount);
+
+                let mut output_pair = LockedAmountAttributesPair {
+                    token_amount: first_payment.amount,
+                    attributes: first_token_attributes,
+                };
+                for payment in &payments {
+                    let attributes: LockedTokenAttributes<Self::Api> =
+                        locked_token_mapper.get_token_attributes(payment.token_nonce);
+                    energy.update_after_unlock_any(
+                        &payment.amount,
+                        attributes.unlock_epoch,
+                        current_epoch,
+                    );
+
+                    locked_token_mapper.nft_burn(payment.token_nonce, &payment.amount);
+
+                    let amount_attr_pair = LockedAmountAttributesPair {
+                        token_amount: payment.amount,
+                        attributes,
+                    };
+                    output_pair.merge_with(amount_attr_pair);
+                }
+
+                energy.add_after_token_lock(
+                    &output_pair.token_amount,
+                    output_pair.attributes.unlock_epoch,
+                    current_epoch,
+                );
```

Comment:
The energy added here uses `output_pair.attributes.unlock_epoch` and not the `normalized_unlock_epoch` which is used below in `self.lock_and_send(&caller, simulated_lock_payment, normalized_unlock_epoch);`
Shouldn't the same `normalized_unlock_epoch` be used in both places?